rsx: Use enum class for surface related regs

vlj opened the pull request.

danilaml commented on rpcs3/Emu/RSX/GCM.cpp:
```diff
@@ -773,6 +798,40 @@ std::string rsx::get_method_name(const u32 id)
+	throw EXCEPTION("unknow surface antialiasing format %x", in);
+}
+
+Surface_color_format to_surface_color_format(u8 in)
```
Why do you need such functions? It seems to obfuscate stuff even more. Why not at least use CELL_GCM values here.

vlj replied:
rsx::registers[] contains int values, they need to be converted (enum class are strongly typed).
I can readd the CELL_GCM values but I'd rather have them hidden in GCM.cpp so that everything outside use the enum values instead.

danilaml replied:
twell, if it makes sense and easier to use some other random enums instead of cell gcm enums, that's alright. But why magic numbers in convert functions? It's hard to see if this case 0 is really cell gcm what it used to be or not.

tambry replied:
I agree that it would be more readable, if CELL_GCM_* values were used in the convert functions.

vlj commented:
Added CELL_GCM* in GCM.cpp and rebased.

@tambry the name of current values is capitalized which is the convention used for macro or defines.
And using custom converted values ensures that code outside of GCM.cpp uses the conversion function, which throw if the value is not recognised. Otherwise it could be possible to static_cast<Surface_target>(some_u32); which is valid even if the value is not representable in the given type.
